### ISSUE TITLE
[ContractKit] ChainId error in paramsPopulator.populate method

### ISSUE THREAD

**Issue by pawelnguyen** [celo-org/celo-monorepo#9219]
Hi! Since a couple of days I'm seeing an error in `kit.connection.paramsPopulator.populate(tx)` method in ContractKit.

### Context
I'm using that method to fill in chainId, nonce, gas and gasPrice of a transaction, which then I'm signing and broadcasting

### Expected Behavior

Here's an example that used to work around a week ago:

```js
import { newKit } from "@celo/contractkit";

(async () => {
  const txParams = {
    from: '0x1b4399a7c97ae092fb4ccdc1598b2767ecb79652',
    value: '10000',
    to: '0xf35ed7156babf2541e032b3bb8625210316e2832',
  }

  const kit = newKit('http://localhost:9393');
  console.log(await kit.connection.paramsPopulator.populate(txParams))
})()
```

Expected result:

```js
{
  from: '0x1b4399a7c97ae092fb4ccdc1598b2767ecb79652',
  value: '10000',
  to: '0xf35ed7156babf2541e032b3bb8625210316e2832',
  chainId: 42220,
  nonce: 19064,
  gas: 21055,
  gasPrice: '0x1dcd6500'
}
```

### Current Behavior

Currently I'm getting:
`(node:62177) UnhandledPromiseRejectionWarning: Gas estimation failed: Could not decode transaction failure reason or Error: invalid argument 0: json: cannot unmarshal non-string into Go struct field TransactionArgs.chainId of type *hexutil.Big`

### My findings

So I've been debugging this and I think that the issue is the following:
1. populate method fetches a chainId if it's empty https://github.com/celo-org/celo-monorepo/blob/4b08278d0d2a951b3de1950f26715e256ecafbc4/packages/sdk/connect/src/utils/tx-params-normalizer.ts#L24
2. this chainId is an integer, as defined in web3-core types https://github.com/ChainSafe/web3.js/blob/5df6818ee4a0a30fb64dd3e94d97964826489b7f/packages/web3-core/types/index.d.ts#L153
3. chainId is set and sent to `connection.estimateGas` method https://github.com/celo-org/celo-monorepo/blob/4b08278d0d2a951b3de1950f26715e256ecafbc4/packages/sdk/connect/src/utils/tx-params-normalizer.ts#L32
3. Celo node expects chainId in a hexutil.Big format and throws an error

Here's an example that does work but isn't compliant with type definitions (chainId should be a number, not a string):

```js
import { newKit } from "@celo/contractkit";

(async () => {
  const txParams = {
    from: '0x1b4399a7c97ae092fb4ccdc1598b2767ecb79652',
    value: '10000',
    to: '0xf35ed7156babf2541e032b3bb8625210316e2832',
    chainId: '0x42220'
  }

  const kit = newKit('http://localhost:9393');
  console.log(await kit.connection.paramsPopulator.populate(txParams))
})()
```

I dug a bit around latest release of celo-blockchain, but the only relevant thing I found is this comment https://github.com/celo-org/celo-blockchain/pull/1791#issuecomment-1011209303

Celo archive node I'm using is running `{"jsonrpc":"2.0","id":1,"result":"celo/v1.5.0-stable-0f7d2e5c/linux-amd64/go1.17.3"}`

**Comment by gastonponti** [comment]
@pawelnguyen
A little explanation:
The chainId was not a field that the node was taking in count. So basically it was not using that field. The Contract Kit, uses the chainId field for adding it to the `v` of the signature to avoid a reply attack (https://eips.ethereum.org/EIPS/eip-155). Signs the tx, and then removes that `chainId` field. And that signature with the tx without the `chainId` is sent to the node.
With the typed transaction envelope (https://eips.ethereum.org/EIPS/eip-2718) new types of txs are added (eip1559, eip2930), which make the chainId field as a valid one, so now the node, checks that field as a valid one. And, basically, expects an hexa "0x1234".

The error is that the CK for the populate, is using a number, instead of a string. It's not a current problem for all the other cycles because all the wallets from the CK basically remove that chainId, but it is a problem for those who use the `populate` method and don't remove that field.

TL;DR; the CK should change that chainId from a number to a string that represents an hexa
(adding @dckesler)

**Comment by pawelnguyen** [comment]
Thanks a lot for taking a look and for the explanation @gastonponti, it makes sense.

Just in case someone stumbles upon this, I worked around it by fetching chainId, nonce, gas and GasPrice manually in a way chainId is not passed to gas estimation:
```js
  const txParams = {
    from: account.freshAddress,
    to: address,
    value
  };

  return {
    ...txParams,
    chainId: await kit.connection.chainId(),
    nonce: await kit.connection.nonce(account.freshAddress),
    gas: await kit.connection.estimateGas(txParams),
    gasPrice: await kit.connection.gasPrice(),
  }
 ```
